qiskit_experiments/calibration_management/parameter_value.py

```diff
@@ -71,6 +72,19 @@ def __post_init__(self):
 
         self.date_time = self.date_time.astimezone()
 
+        if isinstance(self.value, complex):
+            warnings.warn(
```

Comment:
Do we want to start using new decorator in terra? This will block merging this PR until 0.24rc1 release but the release should be end of this month.

Reply:
Your call. Let me know what you prefer.

Reply:
I prefer using new decorator since this is going to be standard. But of course it's no the case if fix is urgent.

Reply:
Following a private communication, we decided to use the current deprecation style because the decorators don't seem to work well with `DataClass`.